lib/AST/ASTMangler.cpp

```diff
@@ -3062,7 +3062,8 @@ void ASTMangler::appendAnyGenericType(const GenericTypeDecl *decl,
       // `ClassTemplateSpecializationDecl`'s name does not include information about
       // template arguments, and in order to prevent name clashes we use the
       // name of the Swift decl which does include template arguments.
-      appendIdentifier(nominal->getName().str());
+      appendIdentifier(nominal->getName().str(),
+                       /*allowRawIdentifiers=*/false);
```

Comment:
I feel like we should revisit the naming of imported class template specializations in light of raw identifiers, but I don't want to bring any of that discussion into this pull request.